force-app/main/default/classes/PermissionSetAssignmentMapper_TEST.cls

```diff
@@ -0,0 +1,137 @@
+    * @description Test method to verify that the getInstance method returns a new instance of the 
+    * PermissionSetAssignmentMapper class when one does not already exist.
+    ***************************************************************************************************************************/
+    @isTest
+    private static void getInstanceNew() {
+        Test.startTest();
+        PermissionSetAssignmentMapper permissionSetAssignmentMapperInstance = PermissionSetAssignmentMapper.getInstance();
+        Test.stopTest();
+
+        System.assertEquals(PermissionSetAssignmentMapper.instance, permissionSetAssignmentMapperInstance, 'Instance of mapper class returned should match static instance variable.');
+        System.assertEquals(true, permissionSetAssignmentMapperInstance != null, 'Instance of mapper class should not be null.');
+    }
+
+    /**************************************************************************************************************************
+    * @description Test method to verify that the getInstance method returns the existing instance of the 
+    * PermissionSetAssignmentMapper class when one already exists.
+    ***************************************************************************************************************************/
+    @isTest
+    private static void getInstanceExisting() {
+        Test.startTest();
+            PermissionSetAssignmentMapper permissionSetAssignmentMapperInstance1 = PermissionSetAssignmentMapper.getInstance();
+            PermissionSetAssignmentMapper permissionSetAssignmentMapperInstance2 = PermissionSetAssignmentMapper.getInstance();
+        Test.stopTest();
+
+        System.assertEquals(PermissionSetAssignmentMapper.instance, permissionSetAssignmentMapperInstance1, 'Instance of mapper class returned should match static instance variable.');
+        System.assertEquals(permissionSetAssignmentMapperInstance1, permissionSetAssignmentMapperInstance2, 'Subsequent retrievals of mapper class instance should return existing instance.');
+        System.assertEquals(true, permissionSetAssignmentMapperInstance1 != null, 'Instance of mapper class should not be null.');
+    }
+
+    /**************************************************************************************************************************
+    * @description Test method to verify that the getAssignedPermissionSetIdSetForCurrentUser method 
+    * returns a set of permission set ids assigned to the current user when there are some assigned
+    ***************************************************************************************************************************/
+    @isTest
+    private static void getAssignedPermissionSetIdsSetForCurrentUserExisting() {
+        User user = UTIL_UnitTestData_TEST.createNewUserForTests('testUser@eda.com');
+
+        List<PermissionSet> permissionSets = new List<PermissionSet>();
```

Comment:
This is an elegant solution to testing! I did not know you could do that!

Reply:
I just found out too!